src/spicy/spicyz/CMakeLists.txt

```diff
@@ -10,7 +10,18 @@ set_target_properties(spicyz PROPERTIES CXX_EXTENSIONS OFF)
 
 target_include_directories(spicyz PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 
-target_link_libraries(spicyz PRIVATE hilti spicy)
+if (BINARY_PACKAGING_MODE)
```

Comment:
Just commenting for the record, nothing to change.

I'm somewhat confused by this (and have been a bit before with https://github.com/zeek/zeek/pull/3565)

This seems to assume BINARY_PACKAGING_MODE never uses an externally provided Spicy and it's possible to link against the in-tree objects. The other thought, are we just linking object files into `spicyz` to full-fill some "rpaths-not-allowed" rule? Seems like a dishonest solution to that :-)

Fedora documents it's okay to use RPATH for internal libraries that are used by executables to enable code sharing. Could term hilti/spicy as such IMO.
https://fedoraproject.org/wiki/RPath_Packaging_Draft#Rpath_for_Internal_Libraries